include/circt/Support/PrettyPrinterHelpers.h

```diff
@@ -166,6 +166,20 @@ class TokenStringSaver : public PrettyPrinter::Listener {
   void clear() override;
 };
 
+class TokenStringAndCallbackSaver : public TokenStringSaver {
+  llvm::SpecificBumpPtrAllocator<Token::CallbackInfo::CallbackTy> callbackAlloc;
```

Comment:
This is concerning.  The primary reason to use such an allocator is to defer lifetime operations and to handle high allocation rates.  How many unique callbacks are there really?

Reply:
There is a unique callback for each op that is printed, because the callback lambda also captures the op.